### PR TITLE
[collect] refactor _format_version

### PR DESCRIPTION
**Pull request by dnegreira**
Ensure that we format the package versions to the pep440 standard. This is necessary as we are using `parse_version()` from the `packaging` package to verify that the sos collector node version is equal or bigger than the versions we are collecting from the nodes. If we pass the wrong format to `parse_version()`, we are unable to do the comparison and unable to run `sos report` on those nodes.
This addresses the issue of the Ubuntu packaging version naming with `+` and `~` signals as well as generic versioning schemes.

Resolves: #3544 
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by dnegreira** [comment]
I have decided to refactor the whole _format_version method as it was not compatible with the pep440 standard.
I deviated of what I have initially writen on #3544, and I am not just substituting the `~` and `+` characters for a `.`, but making the format compatible with pep440.

Since I am re-writing / deprecating a whole method, I am happy to hear any kind of feedback, especially from @TurboTurtle as he has written this function in particular.

From my quick investigation with CentOS8, the package policy for `rpm` will always be compatible with the format since the version returned is the major version IE, `4.X`. But appreciate some testing from anyone who can do it from RedHat or any other distro.

Talking about Testing, I think some unit testing would be -super- useful in this case and I am willing to spend some cycles on doing it, either on this PR or on a new PR, but I would need a bit of handholding as I have never really done anything with avocado, and setting up the basic infrastructure so that we can mock and do all the neat things with collect.

File: sos/collector/sosnode.py [hunk 1]
```diff
@@ -412,25 +412,29 @@ def check_sos_version(self, ver):
...
-                    _rel = '-' + ver.split('-')[-1].split('.')[0]
-                if len(_fver.split('.')) == 2:
-                    _fver += '.0'
-
-                return _fver + _rel
+                _, public, local = public_version_re.split(ver, maxsplit=1)
+                if not local:
+                    return ver
+                sanitized_local = re.sub("[+~]+", ".", local).strip("-")
+                pep440_version = "%s+%s" % (public, sanitized_local)
```
Comment (TurboTurtle) [inline sos/collector/sosnode.py:425]: f-string here please.

File: sos/collector/sosnode.py [hunk 2]
```diff
@@ -412,25 +412,29 @@ def check_sos_version(self, ver):
...
+                return pep440_version
             except Exception as err:
-                self.log_debug("Unable to format '%s': %s" % (ver, err))
+                self.log_debug("Unable to format '%s' to pep440 format: %s"
+                               % (ver, err))
```
Comment (TurboTurtle) [inline sos/collector/sosnode.py:429]: Here too.

**Review by TurboTurtle — COMMENTED** [review]
Overall looks good to me, minor comments below.

File: sos/collector/sosnode.py (hunk 1 above)
Comment (dnegreira) [reply sos/collector/sosnode.py]: Whooops, gotta remind myself of this now :)

fixed on: [fceb5f3](https://github.com/sosreport/sos/commit/fceb5f393bbeb5c9da1fa63877a2f8d2294f223f)

File: sos/collector/sosnode.py (hunk 2 above)
Comment (dnegreira) [reply sos/collector/sosnode.py]: fixed on: [fceb5f3](https://github.com/sosreport/sos/commit/fceb5f393bbeb5c9da1fa63877a2f8d2294f223f)

**Review by TurboTurtle — APPROVED** [review]

**Review by pmoravec — APPROVED** [review]
It is strange to see versions like `4.6.0-5` to be formatted to `4.6.0+5` but it really follows the [PEP specification](https://peps.python.org/pep-0440/#local-version-identifiers).

**Pull request merged**